src/main/java/com/uber/cadence/internal/worker/LocalActivityWorker.java

```diff
@@ -0,0 +1,248 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.worker;
+
+import com.uber.cadence.*;
+import com.uber.cadence.common.RetryOptions;
+import com.uber.cadence.converter.JsonDataConverter;
+import com.uber.cadence.internal.metrics.MetricsType;
+import com.uber.cadence.internal.replay.ClockDecisionContext;
+import com.uber.cadence.internal.replay.ExecuteLocalActivityParameters;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.LongSupplier;
+
+public final class LocalActivityWorker implements SuspendableWorker {
+
+  private static final String POLL_THREAD_NAME_PREFIX = "Local Activity Poller taskList=";
+
+  private SuspendableWorker poller = new NoopSuspendableWorker();
+  private final ActivityTaskHandler handler;
+  private final String domain;
+  private final String taskList;
+  private final SingleWorkerOptions options;
+  private final LocalActivityPollTask laPollTask;
+
+  public LocalActivityWorker(
+      String domain, String taskList, SingleWorkerOptions options, ActivityTaskHandler handler) {
+    this.domain = Objects.requireNonNull(domain);
+    this.taskList = Objects.requireNonNull(taskList);
+    this.options = options;
+    this.handler = handler;
+    this.laPollTask = new LocalActivityPollTask();
+  }
+
+  @Override
+  public void start() {
+    if (handler.isAnyTypeSupported()) {
+      PollerOptions pollerOptions = options.getPollerOptions();
+      if (pollerOptions.getPollThreadNamePrefix() == null) {
+        pollerOptions =
+            new PollerOptions.Builder(pollerOptions)
+                .setPollThreadNamePrefix(
+                    POLL_THREAD_NAME_PREFIX
+                        + "\""
+                        + taskList
+                        + "\", domain=\""
+                        + domain
+                        + "\", type=\"activity\"")
+                .build();
+      }
+      poller =
+          new Poller<>(
```

Comment:
I'm a little confused about how threads are allocated for local activity execution. Is pool size for them separately configurable?

Reply:
Added setMaxConcurrentLocalActivityExecutionSize in WorkerOption.